lib/crawlexir/search.ex

```diff
@@ -76,61 +76,67 @@ defmodule Crawlexir.Search do
     Ecto.Multi.new()
     |> Ecto.Multi.run(:keyword, fn _, _ -> create_keyword(user, attrs) end)
     |> Ecto.Multi.run(:worker, fn _, %{keyword: keyword} ->
-      scrap_results_for_keyword(keyword.id)
+      create_scrap_worker_job(keyword.id)
     end)
     |> Repo.transaction()
   end
 
+  defp create_scrap_worker_job(keyword_id) do
+    %{keyword_id: keyword_id}
+    |> ScraperWorker.new()
+    |> Oban.insert()
+  end
+
   @doc """
-  Updates a keyword.
+  Parse keywords list from a CSV file.
+  Parsing relies on the provided template stored in "/static/csv/template.csv"
 
   ## Examples
 
-      iex> update_keyword(keyword, %{field: new_value})
-      {:ok, %Keyword{}}
-
-      iex> update_keyword(keyword, %{field: bad_value})
-      {:error, %Ecto.Changeset{}}
+      iex> "./assets/static/csv/template.csv"
+      iex> |> Path.expand(__DIR__)
+      iex> |> Crawlexir.Search.parse_keyword_file
+      {:ok, ["first_keyword", "second_keyword"]}
 
   """
-  def update_keyword(%Keyword{} = keyword, attrs) do
-    keyword
-    |> Keyword.changeset(attrs)
-    |> Repo.update()
+  def parse_keyword_file(file) do
+    Csv.parse(file)
   end
 
   @doc """
-  Returns an `%Ecto.Changeset{}` for tracking keyword changes.
+  Get a keyword report.
 
   ## Examples
 
-      iex> change_keyword(keyword)
-      %Ecto.Changeset{source: %Keyword{}}
+      iex> get_keyword_report(keyword_id)
+      {:ok, %Report{}}
+
+      iex> get_keyword_report(keyword_id)
+      nil
 
   """
-  def change_keyword(%Keyword{} = keyword) do
-    Keyword.changeset(keyword, %{})
+  def get_keyword_report(keyword_id) do
+    Report
+    |> where(keyword_id: ^keyword_id)
+    |> preload(:keyword)
+    |> Repo.one()
```

Comment:
Just wondering if this produce the same thing as `Repo.get_by(Keyword, keyword_id: keyword_id)`?

Reply:
`Repo.get_by(Report, keyword_id: keyword_id)` would fetch the right record but not reload the associated keyword. Then I would need to do pipe preload:

```elixir
Repo.get_by(Report, keyword_id: keyword_id) |> Repo.preload(:keyword)
```

Personally, I find the current query piping nicer as it read like `Report.where().preload()`. What do you think?

Reply:
Yep 👍